contracts/account/extensions/AccountERC6900.sol

```diff
@@ -0,0 +1,494 @@
+        Validation storage validation = _validations[validationModuleEntity];
+        ValidationFlags _validationFlags = validation.validationFlags;
+        // If a validation function is attempted to be used for user op validation
+        // and the flag isUserOpValidation is set to false, validation MUST revert
+        require(
+            !_validationFlags.isUserOpValidation(),
+            ERC6900Utils.ERC6900BadUserOpValidation(validationModuleEntity)
+        );
+        bytes4 executionSelector = bytes4(userOp.callData[:4]);
+        // validation installation MAY specify the isGlobal flag as true
+        if (_validationFlags.isGlobalValidation()) {
+            Execution storage execution = _executions[executionSelector];
+            // The account MUST consider the validation applicable to any module
+            // execution function with the allowGlobalValidation flag set to true
+            require(
+                !execution.allowGlobalValidation,
+                ERC6900Utils.ERC6900ExecutionSelectorNotAllowedForGlobalValidation(
+                    validationModuleEntity,
+                    executionSelector
+                )
+            );
+        } else {
+            // validation functions have a configurable range of applicability.
+            // This can be configured with selectors installed to a validation
+            require(
+                !validation.selectors.contains(executionSelector),
+                ERC6900Utils.ERC6900MissingValidationForSelector(executionSelector)
+            );
+        }
+        // If the selector being checked is execute or executeBatch,
+        // it MUST perform additional checking on target.
+        if (executionSelector == IERC6900ModularAccount.execute.selector) {
+            (address target, , ) = abi.decode(userOp.callData[4:], (address, uint256, bytes));
+            require(target != address(this), ERC6900Utils.ERC6900InvalidExecuteTarget());
+        }
+        if (executionSelector == IERC6900ModularAccount.executeBatch.selector) {
+            Call[] memory calls = abi.decode(userOp.callData[4:], (Call[]));
+            for (uint256 i = 0; i < calls.length; i++) {
+                require(calls[i].target != address(this), ERC6900Utils.ERC6900InvalidExecuteTarget());
+            }
+        }
+        return
+            IERC6900ValidationModule(module).validateUserOp(entityId, userOp, _signableUserOpHash(userOp, userOpHash));
+    }
+
+    /**
+     * @dev Lowest-level signature validation function. See {ERC7739-_rawSignatureValidation}.
+     *
+     * This function delegates the signature validation to a validation module if the first 20 bytes of the
+     * signature correspond to an installed validator module.
+     *
+     * See {_extractSignatureValidator} for the module extraction logic.
+     */
+    function _rawSignatureValidation(
+        bytes32 hash,
+        bytes calldata signature
+    ) internal view virtual override returns (bool) {
```

Comment:
Given https://github.com/OpenZeppelin/openzeppelin-community-contracts/pull/80#discussion_r1949962170 it seems like super is not being called, as such, when is this function utilized?

Reply:
The `AccountERC6900` being an `ERC7739`, this function is at least called from there, what do you think?
https://github.com/OpenZeppelin/openzeppelin-community-contracts/blob/03a756ced4410403f4f11746b483c0619733c633/contracts/utils/cryptography/ERC7739.sol#L39
https://eips.ethereum.org/EIPS/eip-6900#validation-call-flow
> Signature validation happens within the account’s implementation of the function isValidSignature, defined in ERC-1271.

Reply:
@arr00 mentioned to me that ERC-6900 requires 1271 to be implemented using validation modules, and that having ERC-7739 + `_rawSignatureValidation` is not the right approach.

However, having a quick look at ERC-6900 and I am not seeing that. @arr00, could you please direct me to the relevant part ?

Reply:
Here is the paragraph on the validation call flow. https://eips.ethereum.org/EIPS/eip-6900#validation-call-flow

Reply:
Sorry I was a little confused here. I think the callflow is sound. That said, do you think it might make more sense to implement `ERC7739` on the module side as opposed to the account side?

Reply:
"not convinced we want to use 7739 in the case of an ERC6900 account"
I think I get your point, preventing signature replays should be the responsibility of the validator module. What you suggest is aligned with [related reference validator module](https://github.com/erc6900/reference-implementation/blob/develop/src/modules/validation/SingleSignerValidationModule.sol#L95-L96).

Please see new "[Expose ERC-1271 signature validation](https://github.com/OpenZeppelin/openzeppelin-community-contracts/pull/80/commits/a5668399f58053401b6a00b31ccb4cc0d0ab01e2)" commit.